datafusion/core/src/physical_optimizer/utils.rs

```diff
@@ -45,3 +45,74 @@ pub fn optimize_children(
         with_new_children_if_necessary(plan, children)
     }
 }
+
+/// Apply transform `F` to the plan's children, the transform `F` might have a direction(Preorder or Postorder)
+fn map_children<F>(
```

Comment:
How is this code different than what is in datafusion/physical-expr/src/utils.rs ?

Reply:
Yes, those are duplications, those utils are applied to Arc<dyn ExecutionPlan> and the others applied to Arc<dyn PhysicalExpr>.   Maybe I should define a common TreeVisitor struct to hold those methods, but I don't how to make it work for both Arc<dyn ExecutionPlan> and Arc<dyn PhysicalExpr> in Rust.

Need help, any suggestion here ?

Reply:
Ah, I missed the subtlety that they were for (slightly) different classes. 🤔